src/api/gallery.py

```diff
@@ -0,0 +1,127 @@
+"""
+Gallery Endpoints to Fetch, Upload and Remove image from a pro's gallery
+"""
+
+from flask import Blueprint, jsonify, request, abort
+import json
+import uuid
+import datetime
+from hashlib import md5
+import config
+from store import upload_blob, delete_blob
+from utils import load_json, save_json, create_thumbnail
+
+gallery = Blueprint('gallery', __name__)
+
+@gallery.route('/<string:pid>/gallery/', methods=['GET'])
+def fetch_gallery_details(pid):
+    """
+    Fetches gallery details of a professional
+    """
+    # load the database
+    try:
+        db = load_json(config.DB['JSON_PATH'])
+        if not db:
+            abort(404, description='No professionals found')
+    except (FileNotFoundError, PermissionError, json.JSONDecodeError) as e:
+        abort(500, description=str(e))
+
+    # check if the pro exists
+    if pid not in db:
```

Comment:
Check this first before connecting to database.

Reply:
I do not understand how will I be able to check if a pro exists with the passed `pid` without checking it in the database?

Reply:
My bad. I was misleading. This is actually a complex change. What I'm suggesting is to test if "the pro exists" outside the route itself in a decorator. Then just access the gallery using the db connection. For example:

```python
@gallery.route('.......')
@if_pro_exists
def get_gallery(pro):
     # fetch the gallery details
```

The decorator `if_pro_exists` will itself connect to the db and validate that the pro exists.

Also, I would recommend move all the database related code to a separate file `db.py`.
Also, drop the blueprints and move all the routing code to `api.py`. So, that someone who wants to know what are all the routes available in the api can check there. `api.py` would import the individual module like `gallery.py` and `pro.py` and import functions from those module and assign them to the routes.